syntax = "proto3";

package block_chain;

message BlockProto {
  repeated TransactionProto transactions = 1;
  int64 timestamp = 2;
  bytes hash = 3;
  bytes prev_hash = 4;
  int64 nonce = 5;
  bool is_valid = 6;
}

message TransactionProto {
  string tx_id = 1;
  repeated TransactionInputProto inputs = 2;
  repeated TransactionOutputProto outputs = 3;
}

message TransactionInputProto {
  string tx_id = 1;
  int64 vout = 2;
  string script_sig = 3;
}

message TransactionOutputProto {
  string address = 1;
  string amount = 2;
}